pkg/operation/operation.go

```diff
@@ -83,9 +83,13 @@ func newOperation(
 	if err != nil {
 		return nil, err
 	}
-	seedObj, err := seed.NewFromName(k8sGardenClient, k8sGardenInformers, seedName)
-	if err != nil {
-		return nil, err
+
+	var seedObj *seed.Seed
+	if seedName != nil {
```

Comment:
I guess we should prevent instantiating a new operation if the seed name is not set?

Reply:
Yep, that is the case in in shoot_control shoot_care_control and shoot_maintenance control. So i think you are right.

Reply:
Usually, when the seed is not set, we do not reconcile. However in the case of deletion, before the Spec.Cloud.Seed is set, we get to this point & then accept the deletion later.

Reply:
I don't get your last comment. Can you explain again?

Reply:
Here [in shoot_control](https://github.com/gardener/gardener/blob/c4f251b154b94de2443bc068c28aea8f6e4c681d/pkg/controllermanager/controller/shoot/shoot_control.go#L166) if the shoot does not have a seed, mayReconcile = false, thus the shoot is not being reconciled. No new operation will be created. We wait for the scheduler to do its work.

However in the case the shoot should be deleted we always need to reconcile. See [here](https://github.com/gardener/gardener/blob/c4f251b154b94de2443bc068c28aea8f6e4c681d/pkg/controllermanager/controller/shoot/shoot_control.go#L161).
In the reconcile [a new operation is created ](https://github.com/gardener/gardener/blob/c4f251b154b94de2443bc068c28aea8f6e4c681d/pkg/controllermanager/controller/shoot/shoot_control.go#L308). In the case we delete a shoot that has not yet a seed set, it would error -> hence the check above.
Then we can delete the shoot [here](https://github.com/gardener/gardener/blob/c4f251b154b94de2443bc068c28aea8f6e4c681d/pkg/controllermanager/controller/shoot/shoot_control.go#L337). For shoots with no seed set yet, the deletion is being accepted right away, because it has never been reconiled/ len(o.Shoot.Info.Status.UID) == 0.